feat: sign without hash

JuyeonYu opened the pull request:
## **Summary of Changes**

For signing, add bool value using  hash or not .
There is the same bool argument for Web3j for Java.
This PR will help collaborate with Java developers.

yaroslavyaroslav commented:
@JuyeonYu thank for your participation, there's just a few requests about that:

1. Could you please to not recreate PRs, but update the excited ones if it's required. 
2. I speculate that you're confused with the failing ci/cd. It has failed on code styling checked with a swiftlint. Please check the swiftlint config, and try to run it locally to figure what exactly its rule violated within tour pr.

JuyeonYu commented:
Hello. sorry for my insufficient reason and the others.
and for the failing ci/cd, I added default value for the new argument.

nowadays I collaborate with Java developers for making DAPP.
and of course, the differences between web3j for Java and web3swift are being discovered day by day.

In this PR, I added 'useHash' argument in making signature function.
the reason is why in the web3j for Java, there is the argument.
So when they make a signature without hash, we can't verify signature each other.

yaroslavyaroslav commented:
I'm fine with that. What about you folks? @janndriessen @JeneaVranceanu
